### PR TITLE
Status matcher

### PR DESCRIPTION
**Pull request by majk-p**
Solution to: https://github.com/softwaremill/tapir/issues/675 

~WIP due to unsolved mystery of ambiguous signature~

### CODE REVIEW

File: core/src/main/scala/sttp/tapir/Tapir.scala [hunk 1]
```diff
@@ -174,6 +174,34 @@ trait Tapir extends TapirDerivedInputs with ModifyMacroSupport {
...
+    *
+    * Should be used in [[oneOf]] output descriptions.
+    */
+  // def statusMappingExactMatcher[T: ClassTag](
+  //   statusCode: StatusCode,
+  //   output: EndpointOutput[T]
+  // )(
+  //   exactValues: Set[T]
+  // ): StatusMapping[T] =
+  //   statusMappingValueMatcher(statusCode, output)(exactMatch(exactValues))
```
Comment (majk-p) [inline core/src/main/scala/sttp/tapir/Tapir.scala:189]: For a reason I cannot understand, uncommenting this implementation causes ambiguous reference error in `tests/package.scala`:
> [error] /home/majkp/code/tapir/tests/src/main/scala/sttp/tapir/tests/package.scala:198:11: ambiguous reference to overloaded definition,
[error] both method statusMappingExactMatcher in trait Tapir of type [T](statusCode: sttp.model.StatusCode, output: sttp.tapir.EndpointOutput[T])(firstExactValue: T, rest: T*)(implicit evidence$17: scala.reflect.ClassTag[T])sttp.tapir.EndpointOutput.StatusMapping[T]
[error] and  method statusMappingExactMatcher in trait Tapir of type [T](statusCode: sttp.model.StatusCode, output: sttp.tapir.EndpointOutput[T])(exactValues: Set[T])(implicit evidence$16: scala.reflect.ClassTag[T])sttp.tapir.EndpointOutput.StatusMapping[T]
[error] match argument types (sttp.model.StatusCode,sttp.tapir.EndpointIO.Body[String,sttp.tapir.tests.TestAppError])

@kubukoz @adamw - can you please have a look at it?

Comment (kubukoz) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: Apparently overload resolution in Scala 2 works only up to the first parameter list. Couldn't find the reason why it is so, but dotty fixes it - https://dotty.epfl.ch/docs/reference/changed-features/overload-resolution.html

I would suggest using different names for these methods - that'll be the least hacky solution :)

Comment (adamw) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: do we need a `Set` variant? We can always do `(set: _*)`?

Comment (majk-p) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: It's here to address @kubukoz's comment: https://github.com/softwaremill/tapir/issues/675#issuecomment-662387665 
How do you feel about renaming this variant to `statusMappingSetMatcher`? or do you think we should avoid it at all?

Comment (adamw) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: What's the use-case (an example :) )

Comment (majk-p) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: I can't really think of any that's not addressed by the `T*` matcher.

Comment (kubukoz) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: I never wanted to accept `Set[T]`, just `T => Boolean` would be enough as every set is a subtype of that ;P

Comment (majk-p) [reply core/src/main/scala/sttp/tapir/Tapir.scala:189]: Okay, then it's just my misunderstanding. Should we still implement that taking this comment https://github.com/softwaremill/tapir/issues/675#issuecomment-663401150 (need to provide `T` by hand) into account?

Comment (majk-p) [reply core/src/main/scala/sttp/tapir/Tapir.scala]: `Set` based implementation is dropped now.

**Comment by majk-p** [comment]
@adamw I've removed the WIP status, PR is ready for review :wink:

**Comment by adamw** [comment]
Released in 0.16.9, thanks!

**Pull request merged**